dashboard/src/hooks/use-is-changed.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+
+export default function useIsChanged<T extends Record<string, any>>(
+  currentData: T | null | undefined,
+  originalData: T | null | undefined,
+): boolean {
+  const [isChanged, setIsChanged] = useState(false);
+
+  useEffect(() => {
+    if (!originalData || !currentData) {
+      setIsChanged(false);
+      return;
+    }
+
+    const keys = Object.keys(currentData) as (keyof T)[];
+    const changes = keys.some((key) => currentData[key] !== originalData[key]);
+    setIsChanged(changes);
+  }, [currentData, originalData]);
+
+  return isChanged;
```

Comment:
Nitpick: This can be simplified by the use of `useMemo` 😄 
```suggestion
  return useMemo(() => {
    if (!originalData || !currentData) {
      return false;
    }

    const keys = Object.keys(currentData) as (keyof T)[];
    return keys.some((key) => currentData[key] !== originalData[key]);
  }, [currentData, originalData]);
```